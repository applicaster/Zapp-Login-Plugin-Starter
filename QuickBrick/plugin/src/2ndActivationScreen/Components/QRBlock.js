import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  ActivityIndicator
} from 'react-native';
import QRCode from './QRCode';

function QRBlock({ loading, QRCodehintStyle, qrCodeHint, activationCodeUrl }) {
  return (
    loading
      ? (
        <View style={styles.loadContainer}>
          <ActivityIndicator size="large" color="#525A5C" />
        </View>
      )
      : (
        <View style={{ alignItems: 'center' }}>
          <QRCode url={activationCodeUrl} />
          <Text
            style={{ ...QRCodehintStyle, marginTop: 20 }}
            numberOfLines={2}
            ellipsizeMode="tail"
          >
            {qrCodeHint}
          </Text>
        </View>
      )
  );
}

export default QRBlock;

const styles = StyleSheet.create({
  loadContainer: {
    width: 300,
    height: 300,
    justifyContent: 'center',
    alignItems: 'center'
  }
});
