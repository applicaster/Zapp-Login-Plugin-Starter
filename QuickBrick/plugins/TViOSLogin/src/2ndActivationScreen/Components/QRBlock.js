import React from 'react';
import {
  View,
  Text,
  ActivityIndicator
} from 'react-native';
import QRCode from './QRCode';

function QRBlock({ loading, qrCodeHintStyle = {}, qrCodeHint = '', activationCodeUrl = '' }) {
  return (
    <View style={styles.container}>
      {
        loading
          ? <ActivityIndicator size="large" color="#D8D8D8" />
          : (
            <View style={{ alignItems: 'center' }}>
              <QRCode url={activationCodeUrl} />
              <Text
                style={{ ...qrCodeHintStyle, marginTop: '3.5%' }}
                numberOfLines={1}
                ellipsizeMode="tail"
              >
                {qrCodeHint}
              </Text>
            </View>
          )
      }
    </View>
  );
}

export default QRBlock;

const styles = {
  container: {
    minWidth: 340,
    minHeight: 340,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: '6.3%',
    marginRight: '5.5%'
  }
};
