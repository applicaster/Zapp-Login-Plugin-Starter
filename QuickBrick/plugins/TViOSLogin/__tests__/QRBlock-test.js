import React from 'react';
import renderer from 'react-test-renderer';
import QRBlock from '../src/2ndActivationScreen/Components/QRBlock';

describe('QRBlock', () => {
  it('renders correctly', () => {
    const defaultProps = {
      loading: false,
      qrCodeHintStyle: {},
      qrCodeHint: '',
      activationCodeUrl: ''
    };
    const tree = renderer
      .create(<QRBlock props={defaultProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('check loading indicator displayed', () => {
    const props = {
      loading: true
    };

    const testRenderer = renderer.create(<QRBlock props={props} />);
    const testInstance = testRenderer.root;

    const spinner = testInstance.findAllByType('ActivityIndicator')[0];
    expect(spinner);
  });
});
