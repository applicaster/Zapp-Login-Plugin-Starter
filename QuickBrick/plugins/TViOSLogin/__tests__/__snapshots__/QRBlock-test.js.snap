// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRBlock renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginRight": "5.5%",
      "marginTop": "6.3%",
      "minHeight": 340,
      "minWidth": 340,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "height": 340,
          "width": 340,
        }
      }
    />
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "marginTop": "3.5%",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
